lake/index/index.go

```diff
@@ -108,67 +107,24 @@ func (i Index) Proc() (ast.Proc, error) {
 	case IndexZed:
 		return i.zqlProc()
 	default:
-		return nil, fmt.Errorf("unknown index kind: %s", i.Kind)
+		return "", fmt.Errorf("unknown index kind: %s", i.Kind)
 	}
 }
 
-var keyName = field.New("key")
-
-var keyAst = ast.Assignment{
-	Kind: "Assignment",
-	LHS:  ast.NewDotExpr(keyName),
-	RHS:  ast.NewDotExpr(keyName),
-}
-var countAst = ast.NewAggAssignment("count", nil, nil)
+const keyName = "key"
 
 // NewFieldRule creates an index that will index all fields of
 // the type passed in as argument.
-func (i Index) typeProc() (ast.Proc, error) {
-	return &ast.Sequential{
-		Kind: "Sequential",
-		Procs: []ast.Proc{
-			&ast.TypeSplitter{
-				Kind:     "TypeSplitter",
-				Key:      keyName,
-				TypeName: i.Value,
-			},
-			&ast.Summarize{
-				Kind: "Summarize",
-				Keys: []ast.Assignment{keyAst},
-				Aggs: []ast.Assignment{countAst},
-			},
-			&ast.Sort{
-				Kind: "Sort",
-				Args: []ast.Expr{ast.NewDotExpr(keyName)},
-			},
-		},
-	}, nil
+func (i Index) typeProc() (string, error) {
+	return fmt.Sprintf("explode this by %s as %s | count() by %s | sort %s", i.Value, keyName, keyName, keyName), nil
 }
 
-func (i Index) fieldProc() (ast.Proc, error) {
-	return &ast.Sequential{
-		Kind: "Sequential",
-		Procs: []ast.Proc{
-			&ast.FieldCutter{
-				Kind:  "FieldCutter",
-				Field: field.Dotted(i.Value),
-				Out:   keyName,
-			},
-			&ast.Summarize{
-				Kind: "Summarize",
-				Keys: []ast.Assignment{keyAst},
-				Aggs: []ast.Assignment{countAst},
-			},
-			&ast.Sort{
-				Kind: "Sort",
-				Args: []ast.Expr{ast.NewDotExpr(keyName)},
-			},
-		},
-	}, nil
+func (i Index) fieldProc() (string, error) {
+	return fmt.Sprintf("cut %s=%s | count() by %s | sort %s", keyName, i.Value, keyName, keyName), nil
 }
 
-func (i Index) zqlProc() (ast.Proc, error) {
-	return compiler.ParseProc(i.Value)
+func (i Index) zqlProc() (string, error) {
+	return i.Value, nil
```

Comment:
Nit: Now that these methods are so simple, maybe just merge them into `zedQuery`?